### PR TITLE
no match page and socket closing

### PR DESCRIPTION
**Pull request by jeffreyquiambao**
- Implemented the no match page based on @dlhxu redirect changes
- addressed memory leak bug that prevented different components from listening to events properly

image

### CODE REVIEW

File: wewatch-client/src/components/noMatch/noMatch.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React, { useState  } from "react";
+import { Redirect } from "react-router-dom";
+
+import './noMatch.css';
+
+const NoMatch = () => {
```
Comment (rosakng) [inline wewatch-client/src/components/noMatch/noMatch.js:6]: can we get a screenshot in the PR of what this page looks like? :octocat:

Comment (jeffreyquiambao) [reply wewatch-client/src/components/noMatch/noMatch.js:6]: done!

File: wewatch-client/src/components/Lobby/Lobby.js [hunk 2]
```diff
@@ -72,8 +72,10 @@ const Lobby = ({location}) => {
             console.log(top10)
             setGoSwipe(true);
         });
+        return () => {socket.off('sessionMembers')};
```
Comment (rosakng) [inline wewatch-client/src/components/Lobby/Lobby.js:75]: thanks for adding this! this is definitely contributing to good practice as we move forward with using sockets! 💯

**Review by rosakng — APPROVED** [review]

**Review by varoongupta — APPROVED** [review]
Thanks for figuring this out :)

**Review by billxsheng — APPROVED** [review]
Nice!

**Review by dlhxu — APPROVED** [review]
Awesome work, this was a huge blocker for our team this sprint. Thanks for taking the time to find a clean fix!

**Pull request merged**